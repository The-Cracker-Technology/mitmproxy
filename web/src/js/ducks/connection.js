export const ConnectionState = {
    INIT: Symbol("init"),
    FETCHING: Symbol("fetching"), // WebSocket is established, but still fetching resources.
    ESTABLISHED: Symbol("established"),
    ERROR: Symbol("error"),
    OFFLINE: Symbol("offline"), // indicates that there is no live (websocket) backend.
}

const defaultState = {
    state: ConnectionState.INIT,
    message: null,
}

export default function reducer(state = defaultState, action) {
    switch (action.type) {

        case ConnectionState.ESTABLISHED:
        case ConnectionState.FETCHING:
        case ConnectionState.ERROR:
        case ConnectionState.OFFLINE:
            return {
                state: action.type,
                message: action.message
            }

        default:
            return state
    }
}

export function startFetching() {
    return { type: ConnectionState.FETCHING }
}

export function connectionEstablished() {
    return { type: ConnectionState.ESTABLISHED }
}

export function connectionError(message) {
    return { type: ConnectionState.ERROR, message }
}
export function setOffline() {
    return { type: ConnectionState.OFFLINE }
}
