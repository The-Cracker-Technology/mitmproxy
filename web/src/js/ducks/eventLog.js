import reduceStore from "./utils/store"
import * as storeActions from "./utils/store"

export const ADD               = 'EVENTS_ADD'
export const RECEIVE           = 'EVENTS_RECEIVE'
export const TOGGLE_VISIBILITY = 'EVENTS_TOGGLE_VISIBILITY'
export const TOGGLE_FILTER     = 'EVENTS_TOGGLE_FILTER'

const defaultState = {
    visible: false,
    filters: { debug: false, info: true, web: true, warn: true, error: true },
    ...reduceStore(undefined, {}),
}

export default function reduce(state = defaultState, action) {
    switch (action.type) {

        case TOGGLE_VISIBILITY:
            return {
                ...state,
                visible: !state.visible
            }

        case TOGGLE_FILTER:
            const filters = { ...state.filters, [action.filter]: !state.filters[action.filter] }
            return {
                ...state,
                filters,
                ...reduceStore(state, storeActions.setFilter(log => filters[log.level]))
            }

        case ADD:
        case RECEIVE:
            return {
                ...state,
                ...reduceStore(state, storeActions[action.cmd](action.data, log => state.filters[log.level]))
            }

        default:
            return state
    }
}

export function toggleFilter(filter) {
    return { type: TOGGLE_FILTER, filter }
}

export function toggleVisibility() {
    return { type: TOGGLE_VISIBILITY }
}

export function add(message, level = 'web') {
    let data = {
        id: Math.random().toString(),
        message,
        level,
    }
    return {
        type: ADD,
        cmd: "add",
        data
    }
}
