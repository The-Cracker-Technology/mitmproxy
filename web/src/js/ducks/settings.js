import { fetchApi } from '../utils'

export const RECEIVE        = 'SETTINGS_RECEIVE'
export const UPDATE         = 'SETTINGS_UPDATE'
export const REQUEST_UPDATE = 'REQUEST_UPDATE'

const defaultState = {

}

export default function reducer(state = defaultState, action) {
    switch (action.type) {

        case RECEIVE:
            return action.data

        case UPDATE:
            return {
                ...state,
                ...action.data,
            }

        default:
            return state
    }
}

export function update(settings) {
    fetchApi.put('/settings', settings)
    return { type: REQUEST_UPDATE }
}
