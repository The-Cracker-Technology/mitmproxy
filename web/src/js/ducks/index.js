import { combineReducers } from "redux"
import eventLog from "./eventLog"
import flows from "./flows"
import settings from "./settings"
import ui from "./ui/index"
import connection from "./connection"
import options from './options'

export default combineReducers({
    eventLog,
    flows,
    settings,
    connection,
    ui,
    options,
})
