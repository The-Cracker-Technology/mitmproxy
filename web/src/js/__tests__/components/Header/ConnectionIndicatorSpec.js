import React from 'react'
import renderer from 'react-test-renderer'
import ConnectedIndicator, { ConnectionIndicator } from '../../../components/Header/ConnectionIndicator'
import { ConnectionState } from '../../../ducks/connection'
import { Provider } from 'react-redux'
import { TStore } from '../../ducks/tutils'

describe('ConnectionIndicator Component', () => {

    it('should render INIT', () => {
        let connectionIndicator = renderer.create(
            <ConnectionIndicator state={ConnectionState.INIT}/>),
            tree = connectionIndicator.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should render FETCHING', () => {
        let connectionIndicator = renderer.create(
            <ConnectionIndicator state={ConnectionState.FETCHING}/>),
            tree = connectionIndicator.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should render ESTABLISHED', () => {
        let connectionIndicator = renderer.create(
            <ConnectionIndicator state={ConnectionState.ESTABLISHED}/>),
            tree = connectionIndicator.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should render ERROR', () => {
        let connectionIndicator = renderer.create(
            <ConnectionIndicator state={ConnectionState.ERROR} message="foo"/>),
            tree = connectionIndicator.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should render OFFLINE', () => {
        let connectionIndicator = renderer.create(
            <ConnectionIndicator state={ConnectionState.OFFLINE} />),
            tree = connectionIndicator.toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should connect to state', () => {
        let store = TStore(),
            provider = renderer.create(
            <Provider store={store}>
                <ConnectedIndicator/>
            </Provider>),
        tree = provider.toJSON()
        expect(tree).toMatchSnapshot()
    })
})

