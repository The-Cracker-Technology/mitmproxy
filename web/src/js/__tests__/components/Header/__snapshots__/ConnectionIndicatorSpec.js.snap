// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionIndicator Component should connect to state 1`] = `
<span
  className="connection-indicator established"
>
  connected
</span>
`;

exports[`ConnectionIndicator Component should render ERROR 1`] = `
<span
  className="connection-indicator error"
  title="foo"
>
  connection lost
</span>
`;

exports[`ConnectionIndicator Component should render ESTABLISHED 1`] = `
<span
  className="connection-indicator established"
>
  connected
</span>
`;

exports[`ConnectionIndicator Component should render FETCHING 1`] = `
<span
  className="connection-indicator fetching"
>
  fetching data…
</span>
`;

exports[`ConnectionIndicator Component should render INIT 1`] = `
<span
  className="connection-indicator init"
>
  connecting…
</span>
`;

exports[`ConnectionIndicator Component should render OFFLINE 1`] = `
<span
  className="connection-indicator offline"
>
  offline
</span>
`;
